server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -116,10 +129,26 @@ public class TaskManager implements ClusterStateApplier {
     private final Map<TcpChannel, ChannelPendingTaskTracker> channelPendingTaskTrackers = ConcurrentCollections.newConcurrentMap();
     private final SetOnce<TaskCancellationService> cancellationService = new SetOnce<>();
 
+    private volatile boolean taskResourceConsumersEnabled;
+    private final Set<Consumer<Task>> taskResourceConsumer;
+
+    public static TaskManager createTaskManagerWithClusterSettings(
```

Comment:
Since there are now two public ways to construct one of these instances, please document the methods and explain why you should use one over the other. If the constructor is public only for testing then please add a comment saying so.

Reply:
I think you can ignore this comment. The method is pretty self describing and I see that the case without clusterSettings is not exclusively in tests.